### PR TITLE
Clarify conda release steps

### PR DESCRIPTION
**Pull request by dorukozturk**
## Motivation

There were unclear parts in the conda release documentation.

### CODE REVIEW

File: docs/source/make_a_release.rst [hunk 1]
```diff
@@ -161,43 +161,71 @@ PyPI: Step-by-step
 Conda: Step-by-step
 -------------------
 
+.. warning::
+
+   Publishing on conda requires you to have corresponding package version uploaded on
+   pypi. So you have to do the pypi release before you do the conda release.
```
Comment (jcfr) [inline docs/source/make_a_release.rst]:
```
on `PyPI`_ and GitHub release before ...
```

File: docs/source/make_a_release.rst [hunk 2]
```diff
@@ -161,43 +161,71 @@ PyPI: Step-by-step
...
 
-      $ cd /tmp && \
-        git clone git@github.com:conda-forge/pynwb-feedstock.git
+      $ cd /tmp && git clone https://github.com/YOURGITHUBUSER/pynwb-feedstock.git
+
+
+4. Download corresponding source for the release version
+
+  .. warning::
+     You have to do the github and pypi release for this step to succeed.
```
Comment (jcfr) [inline docs/source/make_a_release.rst]: I suggest to remove this comment, there is already a warning at the top

Or if we keep it, the wording would have to be updated to be more accurate

File: docs/source/make_a_release.rst [hunk 3]
```diff
@@ -161,43 +161,71 @@ PyPI: Step-by-step
...
+5. Create a new branch
 
    .. code::
 
       $ cd pynwb-feedstock && \
         git checkout -b $release
 
 
-3. Modify meta.yaml
+6. Modify meta.yaml
```
Comment (jcfr) [inline docs/source/make_a_release.rst]:
```
6. Modify ``meta.yaml``
```

**Comment by bendichter** [comment]
looking good so far. Waiting for the CircleCI to go through and the automatic release on pypi

**Comment by jcfr** [comment]
Yeap. As soon as this job is complete, we should have it: https://circleci.com/gh/NeurodataWithoutBorders/pynwb/5961

**Comment by bendichter** [comment]
```
bash-3.2$ mkvirtualenv pynwb-${release}-install-test && \
>   pip install pynwb && \
>   python -c "import pynwb; print(pynwb.__version__)"
bash: mkvirtualenv: command not found
```

**Comment by bendichter** [comment]
```
Bens-MBP:pynwb bendichter$ sed -i "2s/.*/{% set version = \"$release\" %}/" recipe/meta.yaml
sed: -i may not be used with stdin
```

**Comment by jcfr** [comment]
I guess this note at the top of the release document applies to you ...

> That said, considering the instructions below have been tested on a Linux system, they may have to 
> be adapted to work on macOS or Windows.

That said, we should fix the instructions.

**Comment by bendichter** [comment]
ah interesting ok I can probably figure out the difference in syntax here

**Comment by jcfr** [comment]
You could also manually update `meta.yml` replacing the version and the sha256, out attempt to streamline failed you.

**Comment by dorukozturk** [comment]
@bendichter Definitely. I just added those instructions as one liners but all they do is find a string in the meta.yaml file and replacing them. In specific it is setting https://github.com/conda-forge/pynwb-feedstock/blob/master/recipe/meta.yaml#L2-L3 these 2 lines. You can just edit the files manually.

**Comment by dorukozturk** [comment]
@bendichter I added instructions for macOS. See [this](https://github.com/NeurodataWithoutBorders/pynwb/blob/f88066ee75937f241027a427f61c2df2e0cd91ad/docs/source/make_a_release.rst).

**Comment by bendichter** [comment]
I worked through the conda instructions and submitted a PR but I see the old version, 0.3.0 at https://anaconda.org/conda-forge/pynwb. Do I need to merge as well or is it automatic? It says "Only those with write access to this repository can merge pull requests."

**Comment by jcfr** [comment]
Look like conda-forge detected that new version on PyPI. See https://github.com/conda-forge/pynwb-feedstock/pull/2

> submitted a PR but I see the old version

Which PR are you referring to ?

**Comment by bendichter** [comment]
https://github.com/conda-forge/pynwb-feedstock/pull/2

**Comment by dorukozturk** [comment]
I merged it. It should be available soon. Can we merge this PR too?

**Comment by bendichter** [comment]
Yes, sounds good. Can I have write privileges to that repo?

**Comment by jcfr** [comment]
> Yes, sounds good. Can I have write privileges to that repo?

@dorukozturk @oruebel @ajtritt  May be you could help with that ? I do not have admin permission

The repo is this one: https://github.com/conda-forge/pynwb-feedstock

**Comment by dorukozturk** [comment]
I submitted a PR to add all of you but I don't know when it will take effect.

**Pull request merged**